cli/mender-artifact/partition.go

```diff
@@ -49,6 +49,7 @@ type VPImage interface {
 	io.Closer
 	Open(fpath string) (VPFile, error)
 	OpenDir(fpath string) (VPDir, error)
+	dirtyImage()
```

Comment:
Now only repack images when a command has called `dirtyImage()` as a part of it's execution. This means every command like `CopyTo`, `Write`, etc, all need to call this upon execution.

For this PR, I tried plumbing everything from the image wrappers: `imageAndFile` etc.